### DISCUSSION TITLE
Duda sobre JPA, anottations

### DISCUSSION THREAD

**Discussion by Franpanozzo** [dds-jv/foro#311] [TPA - 2021] [answered]
Buenas, tenemos una pequeña duda con respecto al tp anual, si nos pueden dar una mano buenísimo, saludos!

``` java
@Entity
@DiscriminatorValue("D")
public class UsuarioDuenio extends Usuario {

    @OneToMany(mappedBy="usuario", cascade = {CascadeType.ALL}, fetch = FetchType.LAZY)
    List<DatoDeContacto> datoDeContactoList = new ArrayList<>();
```

Cuando vemos la consulta en la base de datos:
![image](https://user-images.githubusercontent.com/62455934/133915161-5b3e8304-b6be-4716-82b0-36e8e9547986.png)

No nos estaría tomando el OneToMany, por lo que tenemos entendido tendría que aparece el usuario_id desde la tabla de DatoDeContacto pero no aparece. Estamos poniendo algo mal en las annotations o algo?? Nos guiamos por el seminario de este año que hace lo mismo

**Comment by gastonprieto** [comment, marked as answer]
Buenas @Franpanozzo!

 el mapped by es para cuando tenes una relacion bidireccional, sirve para indicarle que del otro lado vas a tener los mapeos. En general es conveniente no usar relaciones bidireccionales, porque se complejiza mantenerlas consistentes.

Si despues de sacar el mapped by el problema que te genero una tabla intermedia, lo que tenes que hacer es usar el @JoinColumn y especificar el nombre de la columna en la otra tabla.
